aten/src/TH/generic/THTensorMath.c

```diff
@@ -2828,14 +2828,20 @@ void THTensor_(range)(THTensor *r_, accreal xmin, accreal xmax, accreal step)
 }
 
 void THTensor_(arange)(THTensor *r_, accreal xmin, accreal xmax, accreal step) {
-#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE)
-  int m = fmod(xmax - xmin, step) == 0;
-#else
-  int m = (xmax - xmin) % step == 0;
-#endif
-  if (m)
-    xmax -= step;
-  THTensor_(range)(r_, xmin, xmax, step);
+  ptrdiff_t size;
+  real i = 0;
+
+  THArgCheck(step > 0 || step < 0, 3, "step must be a non-null number");
```

Comment:
Is there a numeric reason why `step != 0` doesn't work?

Reply:
I dont think it really makes sense because if xmin and xmax are different, we can't really step from xmin to xmax with a step of x. If xmin and xmax are the same, how would we choose a size?

Reply:
I think Peter was suggesting that the line read:

```
THArgCheck(step != 0, 3, ...);
```

(not that we allow a zero step)

The error message "step must be a non-null number" is a bit awkward. (I think something like "step must not be 0" will be better).

If you're concerned about `nan`, maybe just add an explicit check for it.

(I realize that this is taken from the `range` code above)

Reply:
Ah okay, that makes sense now. Sorry for the misunderstanding, Peter.

If nobody has any strong objections, I'm going leave this as `step > 0 || step < 0`. I don't think it makes too much of a difference in terms of readability.